engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceVariablesTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.processinstance;
+
+import static io.camunda.zeebe.protocol.record.Assertions.assertThat;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.protocol.record.intent.VariableIntent;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.Map;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class ModifyProcessInstanceVariablesTest {
```

Comment:
❓ I was wondering whether there is a performance cost to this, but I'm not sure actually. Do you know this?

The reason why I wonder, is that each test class adds a static classrule:
```java
@ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
```

I assume that this will be run for each test class. I'd be interested to hear your thoughts about this.

Reply:
Yes, maybe. :thinking:  

I would suggest looking at performance after the abstraction is done and we finally switched to JUnit 5 extension.